x-pack/test/functional/apps/ml/short_tests/settings/calendar_creation.ts

```diff
@@ -96,12 +98,143 @@ export default function ({ getService }: FtrProviderContext) {
+      await asyncForEach(jobConfigs, assignJobToCalendar);
+
+      await ml.settingsCalendar.saveCalendar();
+      await ml.settingsCalendar.assertCalendarRowExists(calendarId);
+
+      await ml.testExecution.logTestStep(
+        'Assert that the calendar is only connected to jobs applied during creation'
+      );
+      await assertOnlyConnectedToJobsAppliedDuringCreation();
+
+      await ml.testExecution.logTestStep(
+        'Assert that the calendar can connect to jobs applied after creation'
+      );
+      await assertConnectedToJobsAppliedAfterCreation();
+    });
+
+    async function assignJobToCalendar(
+      jobConfig: ReturnType<typeof createJobConfig>
+    ): Promise<void> {
+      await ml.settingsCalendar.selectJob(jobConfig.job_id);
+    }
+
+    async function createSingleGroupJobs() {
+      const automatedConfig = createJobConfig('test_calendar_ad_3');
+      const multiMetricConfig = createJobConfig('test_calendar_ad_4');
+      automatedConfig.groups = ['automated'];
+      multiMetricConfig.groups = ['multi-metric'];
+
+      // @ts-expect-error not full interface
+      await asyncForEach(
+        [automatedConfig, multiMetricConfig],
+        async (config) => await ml.api.createAnomalyDetectionJob(config)
+      );
+    }
+
+    async function assertOnlyConnectedToJobsAppliedDuringCreation() {
+      // Assert that the Calendar Management view shows that the calendar is only connected to the jobs
+      // applied at creation
+      const groupsJustAfterCreationVisibleText = await testSubjects.getVisibleText(
+        'mlCalendarListColumnJobs'
+      );
+      expect(groupsJustAfterCreationVisibleText).to.match(/test_calendar_ad_1, test_calendar_ad_2/);
+
+      await ml.navigation.navigateToAnomalyDetection();
+
+      const [testCalendarAdOne, testCalendarAdTwo] = await testSubjects.findAll(
+        'mlJobListRowDetailsToggle'
+      );
+
+      // assert that the created calendar IS applied to test_calendar_ad_1 and test_calendar_ad_2
+      await testCalendarAdOne.click();
+      await testSubjects.existOrFail('mlJobRowDetailsSection-calendars');
+      await testCalendarAdOne.click();
+      await testCalendarAdTwo.click();
+      await testSubjects.existOrFail('mlJobRowDetailsSection-calendars');
+      await testCalendarAdTwo.click();
+
+      const [, , testCalendarAdThree, testCalendarAdFour] = await testSubjects.findAll(
+        'mlJobListRowDetailsToggle'
+      );
+      // assert that the created calendar is NOT applied to test_calendar_ad_3 and test_calendar_ad_4
+      await testCalendarAdThree.click();
+      await testSubjects.missingOrFail('mlJobRowDetailsSection-calendars');
+      await testCalendarAdThree.click();
+      await testCalendarAdFour.click();
+      await testSubjects.missingOrFail('mlJobRowDetailsSection-calendars');
+      await testCalendarAdFour.click();
+    }
+
+    async function assertConnectedToJobsAppliedAfterCreation() {
```

Comment:
With the changes suggested above, I don't think that we need a second assertion method and can just use one combined that navigates to the AD jobs view and asserts calendar assignments there.

Reply:
Roger that, I think [this](https://github.com/elastic/kibana/pull/178553/commits/9900846d0b797d5e5338eed2e4c1db7d9f6a8d0a) does what you are asking for.